ui/components/app/snaps/snap-ui-renderer/components/tooltip.ts

```diff
@@ -0,0 +1,23 @@
+import { JSXElement, Text, TooltipElement } from '@metamask/snaps-sdk/jsx';
+import { getJsxChildren } from '@metamask/snaps-utils';
+import { mapToTemplate } from '../utils';
+import { UIComponentFactory } from './types';
+
+export const tooltip: UIComponentFactory<TooltipElement> = ({
+  element,
+  ...params
+}) => ({
+  element: 'SnapUITooltip',
+  children: getJsxChildren(element).map((children) =>
+    mapToTemplate({ element: children as JSXElement, ...params }),
+  ),
+  propComponents: {
+    content: mapToTemplate({
+      element:
+        typeof element.props.content === 'string'
+          ? Text({ children: element.props.content })
```

Comment:
I wrapped the `content` prop with the `Text` element to be able to pass it directly to the `propComponents` section. This reduces the bloat of having a logic to pass it somewhere else if it's a string. Furthermore we keep a consistent styling.